packages/webpack-cli/lib/groups/runHelp.js

```diff
@@ -90,48 +98,89 @@ const outputHelp = (args) => {
+                    const description = flag.negatedDescription ? flag.negatedDescription : `Negates ${flag.name}`;
+                    return [...allFlags, { name: `no-${flag.name}`, description, type: Boolean }];
+                }, []);
+        };
+
         const title = bold('⬡                     ') + underline('webpack') + bold('                     ⬡');
         const desc = 'The build tool for modern web applications';
         const websitelink = '         ' + underline('https://webpack.js.org');
         const usage = bold('Usage') + ': ' + '`' + green('webpack [...options] | <command>') + '`';
         const examples = bold('Example') + ': ' + '`' + green('webpack help --flag | <command>') + '`';
         const hh = `          ${title}\n\n${websitelink}\n\n${desc}\n\n${usage}\n${examples}`;
-        const output = commandLineUsage([
-            { content: hh, raw: true },
-            {
-                header: 'Available Commands',
-                content: commands.map((cmd) => {
-                    return { name: `${cmd.name} | ${cmd.alias}`, summary: cmd.description };
-                }),
-            },
-            {
-                header: 'Options',
-                optionList: flags
-                    .map((e) => {
-                        if (e.type.length > 1) {
-                            e.type = e.type[0];
-                        }
-
-                        // Here we replace special characters with chalk's escape
-                        // syntax (`\$&`) to avoid chalk trying to re-process our input.
-                        // This is needed because chalk supports a form of `{var}`
-                        // interpolation.
-                        e.description = e.description.replace(/[{}\\]/g, '\\$&');
-
-                        return e;
-                    })
-                    .concat(negatedFlags),
-            },
-        ]);
-
-        logger.raw(output);
+
+        if (args.includes('verbose') || args.includes('--help=verbose')) {
+            flagsToDisplay = [...flags, ...negatedFlags(flags)];
+
+            const headerAndCommands = commandLineUsage([
+                { content: hh, raw: true },
+                {
+                    header: 'Commands',
+                    content: commands.map((cmd) => {
+                        return { name: `${cmd.name} | ${cmd.alias}`, summary: cmd.description };
+                    }),
+                },
+                { header: 'Options', raw: true },
+            ]);
+
+            // print help-header & commands
+            logger.raw(headerAndCommands);
+            // print all options
+            for (const flag of flagsToDisplay) {
+                let flagType;
+                if (Array.isArray(flag.type)) {
+                    // only combo of boolean & string is allowed
+                    flagType = 'boolean, string';
+                } else if (flag.multiple) {
```

Comment:
Multiple can be not only string

Reply:
And multiple types can be `multiple` too, it should be not in `else if`

Reply:
fixed

Reply:
Not supported by webpack yet.

Reply:
It is logic problem and should be fixed even webpack doesn't provide this flags